test/core/templaters/dbt_test.py

```diff
@@ -120,14 +120,82 @@ def test__templater_dbt_templating_test_lex(in_dbt_project_dir, dbt_templater):
 )
 @pytest.mark.dbt
 def test__templated_sections_do_not_raise_lint_error(in_dbt_project_dir, fname):  # noqa
-    """Test that the dbt test has only a new line lint error."""
+    """Test that templated sections in dbt model does not raise a linting error."""
     lntr = Linter(config=FluffConfig(configs=DBT_FLUFF_CONFIG))
     lnt = lntr.lint_path(path="models/my_new_project/" + fname)
     violations = lnt.check_tuples()
     print(violations)
     assert len(violations) == 0
 
 
+@pytest.mark.parametrize(
+    "fname",
+    [
+        "single_trailing_newline.sql",
+    ],
+)
+@pytest.mark.dbt
+def test__single_trailing_newline_at_file_end_does_not_raise_lint_error(
+    in_dbt_project_dir, fname
+):  # noqa
+    """Test that dbt model with a single trailing newline at end of file does not raise a linting error."""
+    lntr = Linter(config=FluffConfig(configs=DBT_FLUFF_CONFIG))
+    lnt = lntr.lint_path(path="models/my_new_project/" + fname)
+    violations = lnt.check_tuples()
+    print(violations)
+    assert len(violations) == 0
+
+
+@pytest.mark.dbt
+def test__dbt_templater_maintains_single_trailing_newline_at_file_end(
+    in_dbt_project_dir, dbt_templater
+):  # noqa
+    """A test to demonstrate that using the dbt templater maintains a single trailing newline at the end of a dbt model."""
+    lexer = Lexer(config=FluffConfig(configs=DBT_FLUFF_CONFIG))
+    templated_file, _ = dbt_templater.process(
+        in_str="",
+        fname="models/my_new_project/single_trailing_newline.sql",
+        config=FluffConfig(configs=DBT_FLUFF_CONFIG),
+    )
+    tokens, lex_vs = lexer.lex(templated_file)
+    assert templated_file.source_str == "select a\nfrom table_a\n"
+    assert templated_file.templated_str == "select a\nfrom table_a\n"
+
+
+@pytest.mark.parametrize(
+    "fname",
+    [
+        "multiple_trailing_newline.sql",
```

Comment:
I think this fixture should be added to https://github.com/sqlfluff/sqlfluff/pull/868/files#diff-5d3be0ba02e85767f3906e01bdb3611ce1c9defc129ba33dde3c2ecd6e6727e0R118 rather than the code being copied